scorecard.js

```diff
@@ -0,0 +1,69 @@
+
+class ScoreCard {
+  //can only have properties in a class, not class consts as inside an object.
+  GUTTER_GAME = [
+    [0, 0],
+    [0, 0],
+    [0, 0],
+    [0, 0],
+    [0, 0],
+    [0, 0],
+    [0, 0],
+    [0, 0],
+    [0, 0],
+    [0, 0],
+  ];
+  
+  PERFECT_GAME = [
+    [10],
+    [10],
+    [10],
+    [10],
+    [10],
+    [10],
+    [10],
+    [10],
+    [10],
+    [10, 10, 10],
+  ];
+  
+  arraysAreIdentical(arr1, arr2){
+    return JSON.stringify(arr1) === JSON.stringify(arr2);
+}
+s
+//JSON.stringify will convert the object into a string
+
+  constructor(frames) {
+    //assignment before if statement only in ruby, not in js
+    if (this.arraysAreIdentical(frames, this.GUTTER_GAME)) {
+      this.result = 0;
+    } else if (frames == this.PERFECT_GAME) {
+      this.result = 300;
+    } else {
+      this.result = this.calculate_scores(frames);
+    }
+  }
+
+  calculate_scores(frames) {
```

Comment:
Looks like a good conversion of the each_with_index method you used in your ruby